packages/runtime/test-runtime-utils/src/mocks.ts

```diff
@@ -782,15 +795,41 @@ export class MockFluidDataStoreRuntime
 	}
 
 	public get isAttached(): boolean {
-		return !this.local;
+		return this.attachState !== AttachState.Detached;
 	}
 
+	private _attachState: AttachState;
 	public get attachState(): AttachState {
-		return this.local ? AttachState.Detached : AttachState.Attached;
+		return this._attachState;
+	}
+	public set attachState(value: AttachState) {
```

Comment:
Why use a new setter here instead of just filling out the existing stubbed `setAttachState()` method?

Reply:
FWIW, a method is generally preferable here since you're kicking off logic rather than a simple set.